apps/desktop/src/lib/trpc/routers/workspaces/workspaces.ts

```diff
@@ -134,11 +135,17 @@ export const createWorkspacesRouter = () => {
 			return db.data.workspaces.slice().sort((a, b) => a.tabOrder - b.tabOrder);
 		}),
 
-		getAllGrouped: publicProcedure.query(() => {
+		getAllGrouped: publicProcedure.query(async () => {
 			const activeProjects = db.data.projects.filter(
 				(p) => p.tabOrder !== null,
 			);
 
+			// Fetch live sandbox statuses
+			const sandboxResult = await cloudApiClient.listSandboxes();
+			const liveSandboxes = new Map(
+				(sandboxResult.sandboxes ?? []).map((s) => [s.id, s.status]),
+			);
+
```

Comment:
this doesnt look good, probably it should be a separate place where this happens